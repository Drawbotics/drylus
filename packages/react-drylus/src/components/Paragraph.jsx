import React from 'react';
import PropTypes from 'prop-types';
import sv from '@drawbotics/drylus-style-vars';
import { css } from 'emotion';


const styles = {
  root: css`
    color: ${sv.colorPrimary};
    line-height: calc(${sv.defaultLineHeight} * 1.5);
  `,
};


const Paragraph = ({ children }) => {
  return <p className={styles.root}>{children}</p>;
};


Paragraph.propTypes = {
  /** Text displayed by the paragraph */
  children: PropTypes.node,
};


export default Paragraph;
