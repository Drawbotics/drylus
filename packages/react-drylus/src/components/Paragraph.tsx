import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Align } from '../enums';
import { Responsive, Style } from '../types';
import { useResponsiveProps } from '../utils';

const styles = {
  root: css`
    color: ${sv.colorPrimary};
    line-height: calc(${sv.defaultLineHeight} * 1.5);
  `,
  alignRight: css`
    text-align: right;
  `,
  alignCenter: css`
    text-align: center;
  `,
};

export interface ParagraphProps {
  /** Text displayed by the paragraph */
  children: React.ReactNode;

  /**
   * @default Align.LEFT
   */
  align?: Align;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Paragraph = ({ responsive, ...rest }: ParagraphProps) => {
  const { children, style, align = Align.LEFT } = useResponsiveProps<ParagraphProps>(
    rest,
    responsive,
  );
  return (
    <p
      style={style}
      className={cx(styles.root, {
        [styles.alignCenter]: align === Align.CENTER,
        [styles.alignRight]: align === Align.RIGHT,
      })}>
      {children}
    </p>
  );
};
