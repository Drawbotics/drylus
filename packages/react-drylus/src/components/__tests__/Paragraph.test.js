import React from 'react';
import { create } from 'react-test-renderer';

import Paragraph, { ParagraphAlign } from '../Paragraph';


describe('Paragraph', () => {
  describe('matches snapshot when', () => {
    it('is aligned to the left', () => {
      const tree = create(
        <Paragraph align={ParagraphAlign.LEFT}>
          Some content
        </Paragraph>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is aligned to the right', () => {
      const tree = create(
        <Paragraph align={ParagraphAlign.RIGHT}>
          Some content
        </Paragraph>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is center aligned', () => {
      const tree = create(
        <Paragraph align={ParagraphAlign.CENTER}>
          Some content
        </Paragraph>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});