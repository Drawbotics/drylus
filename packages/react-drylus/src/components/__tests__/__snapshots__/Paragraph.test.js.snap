// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph matches snapshot when is aligned to the left 1`] = `
.css-kd3s5k-Paragraph__root {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
}

<p
  className="css-kd3s5k-Paragraph__root"
>
  Some content
</p>
`;

exports[`Paragraph matches snapshot when is aligned to the right 1`] = `
.css-162tk1p-Paragraph__root-Paragraph__alignRight {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
  text-align: right;
}

<p
  className="css-162tk1p-Paragraph__root-Paragraph__alignRight"
>
  Some content
</p>
`;

exports[`Paragraph matches snapshot when is center aligned 1`] = `
.css-1a8t6va-Paragraph__root-Paragraph__alignCenter {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
  text-align: center;
}

<p
  className="css-1a8t6va-Paragraph__root-Paragraph__alignCenter"
>
  Some content
</p>
`;
